src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -234,10 +238,75 @@ private async Task IndexAddressesContinuouslyAsync()
                                 AddressIndexerData indexData = this.addressIndexRepository.GetOrCreateAddress(address);
                                 indexData.BalanceChanges.RemoveAll(x => x.BalanceChangedHeight > lastCommonHeader.Height);
                             }
-                        }
-
-                        this.IndexerTip = lastCommonHeader;
 
+                            // Also need to rewind the outpoint repository.
+                            while (this.IndexerTip.HashBlock != lastCommonHeader.HashBlock)
```

Comment:
@zeptin please put this in a new method... something like `RewindOutpointRepository`